### ISSUE TITLE
Android 11 Crash Rate Very High on Google Play

### ISSUE THREAD

**Issue by DarkChibiShadow** [renpy/renpy#4082]
Hello! I've written before about the mobile version specifically and was helped before as well (thank you) but I'm unfortunately here with more reports of crashing. At first, the new fixes seemed to help but as time has gone on, almost 10% of **Android 11** (_specifically_) users seem to be affected.

I tried looking for the specific devices affected today, but Google Play's back end is incredibly unhelpful, so wasn't able to find specific device details but I did find specific trash details. But, I think it being Android 11 specific seems important...

Here's my crash overview since last time:
![25380d7f4149c79d3aa0ed3fa0b0cc3f](https://user-images.githubusercontent.com/77140159/200480193-2539bd18-4475-4bae-ba88-2b3f94354097.png)

Here's that top crash in specific:
![27547fe2ae40e99ecd4d9792b02b3deb](https://user-images.githubusercontent.com/77140159/200480224-c8c8d915-896e-4540-b369-86cd3a24ac9b.png)

![459ba983d282762c9a536b5cc0aec9fe](https://user-images.githubusercontent.com/77140159/200480240-b4d26976-400b-449c-97e6-6d6d0404752b.png)

In general, users usually report that going into any menu (specifically when saving) that the game hard crashes out, or if they minimize the game it crashes entirely.
I have no idea what to do on my end, the game is smaller than it's ever been since I converted most images to webp and cleaned up the files in general.

I'm on the Ren'py 7 track by the way, if somehow all of this is fixed with Ren'py 8 I'll gladly get that too. This has been a persistent issue and once I'm very invested in getting fixed, since by far most of my player base is mobile based and will continue to be in the future-- I know this is the same for many other devs. Please let me know if I'm missing something, I'm here to help.

Thanks so much for your time and for hearing me out again, I've always appreciated your hard work and time.

**Comment by renpytom** [comment]
You can try the 8.1/7.6 nightlies, which include the debugging symbols that can be uploaded to google play to help debug a crash from that version. 

ie, http://nightly.renpy.org/8-nightly-2022-11-07-8.1.0.22110702n/index.html has the android-native-symbols.zip you can upload.

**Comment by DarkChibiShadow** [comment]
> You can try the 8.1/7.6 nightlies, which include the debugging symbols that can be uploaded to google play to help debug a crash from that version.
> 
> ie, http://nightly.renpy.org/8-nightly-2022-11-07-8.1.0.22110702n/index.html has the android-native-symbols.zip you can upload.

Okay! I'll try this and see if I can make it work / what happens, thank you!

**Comment by renpytom** [comment]
It's unlikely to fix the problem directly, but my hope is it'll at least narrow down where the problem is happening (or if it's in all different spots).

**Comment by Andykl** [comment]
After we releases a new version of our game which uses RenPy 8.0.1 (before it was 7.5.1) the amount of crash report increased, especially on Android 11.

I will try to use debugging sybols as well.

**Comment by mattmurdock0010** [comment]
I can't use symbol files on Huawei AppGallery, Huawei does not accept these files in any way. I have a device with this problem my Oppo brand device still crashing on game starting. I also tried symbol files on android studio but i can't get any info, maybe i do something wrong.

Documentation about Huawei's NDK reporting:

https://developer.huawei.com/consumer/en/doc/development/AppGallery-connect-Guides/agc-crash-report-0000001054980565

**Comment by renpytom** [comment]
You should be able to use the symbols that are part of the NDK to upload to huawei, but I don't have plans to support Huawei's store directly.

**Comment by ashl1** [comment]
Hello, Tom
We use 8.0.3 and add Crashlytcics NDK SDK thus we got following crashes in Pre-launch report:

### First

Exception java.lang.RuntimeException: Unable to start activity ComponentInfo{com.justlivesstudio.tags/org.renpy.android.PythonSDLActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.addView(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2913)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3048)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1808)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:193)
  at android.app.ActivityThread.main (ActivityThread.java:6669)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.addView(android.view.View, android.view.ViewGroup$LayoutParams)' on a null object reference
  at org.renpy.android.PythonSDLActivity.onCreate (PythonSDLActivity.java:368)
  at android.app.Activity.performCreate (Activity.java:7136)
  at android.app.Activity.performCreate (Activity.java:7127)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1271)
  at androidx.test.runner.MonitoringInstrumentation.callActivityOnCreate (MonitoringInstrumentation.java:2)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2893)

Google Pixel 2 (virtual)  --- 1080x1920 --- Android 9 (SDK 28) --- x86

----------

### Second:

Exception java.lang.SecurityException: No active admin ComponentInfo{com.google.android.apps.mtaas.deviceadmin/com.google.android.apps.mtaas.deviceadmin.DeviceAdminReceiver}
  at android.os.Parcel.readException (Parcel.java:1958)
  at android.os.Parcel.readException (Parcel.java:1904)
  at android.app.admin.IDevicePolicyManager$Stub$Proxy.isApplicationHidden (IDevicePolicyManager.java:8952)
  at android.app.admin.DevicePolicyManager.isApplicationHidden (DevicePolicyManager.java:7466)
  at com.google.android.apps.mtaas.deviceadmin.HideApps.onStart (HideApps.java:63)
  at android.app.Instrumentation$InstrumentationThread.run (Instrumentation.java:2106)

samsung SM-G960U1 --- 1440x2960 --- Android 8.0 (SDK 26) --- arm64-v8a

----------

### Third

Exception java.lang.RuntimeException: Exception thrown in onCreate() of ComponentInfo{com.google.android.apps.mtaas.deviceadmin/com.google.android.apps.mtaas.deviceadmin.ToggleUserRestriction}: java.lang.SecurityException: No active admin ComponentInfo{com.google.android.apps.mtaas.deviceadmin/com.google.android.apps.mtaas.deviceadmin.DeviceAdminReceiver}
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6050)
  at android.app.ActivityThread.-wrap1
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1764)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6938)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)
Caused by java.lang.SecurityException: No active admin ComponentInfo{com.google.android.apps.mtaas.deviceadmin/com.google.android.apps.mtaas.deviceadmin.DeviceAdminReceiver}
  at android.os.Parcel.readException (Parcel.java:1958)
  at android.os.Parcel.readException (Parcel.java:1904)
  at android.app.admin.IDevicePolicyManager$Stub$Proxy.setUserRestriction (IDevicePolicyManager.java:8542)
  at android.app.admin.DevicePolicyManager.addUserRestriction (DevicePolicyManager.java:7339)
  at com.google.android.apps.mtaas.deviceadmin.ToggleUserRestriction.start (ToggleUserRestriction.java:49)
  at com.google.android.apps.mtaas.deviceadmin.ToggleUserRestriction.onCreate (ToggleUserRestriction.java:22)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6047)

samsung SM-G960U1 --- 1440x2960 --- Android 8.0 (SDK 26) --- arm64-v8a

----------

### Fourth

Your app uses a defective version of the OpenSSL library, which can cause your app to crash. Update to a fixed version of OpenSSL.
Defective versions:
OpenSSL 1.1.1g in lib/armeabi-v7a/librenpython.so

**Comment by renpytom** [comment]
@ashl1 

I've improved the response to "First", though my suspicion is that the underlying cause is an error with the emulator that's preventing SDL from starting propertly, at which point Ren'Py doesn't expect the error and crashes. 

"Second" and "Third" do not seem to involve classes that are present in Ren'Py. They seems spurious. 

"Fourth" has been resolved in the nightlies. 

Everyone: I'm going to close this out. There are a lot of Android changes in Ren'Py 8.1/7.6, as well as a new compiler. I think it will be hard to match these reports with the new version.
